### DISCUSSION TITLE
Bundling source code changes type definitions

### DISCUSSION THREAD

**Discussion by MerlinMason** [colinhacks/zod#2167] [Q&A] [answered]
In my source code if I write...
```ts
const example = z.object({
    name: z.string(),
});
export type Example = z.infer<typeof example>;
```

I can hover `Example` and it confirms
```ts
type Example = {
    name: string;
}
```

However when I bundle my code (using Rollup) it transforms the above into this
```ts
declare const example: z.ZodObject<{
    name: z.ZodString;
}, "strip", z.ZodTypeAny, {
    name?: string;
}, {
    name?: string;
}>;
export type Example = z.infer<typeof example>;
```

In the bundled code if I hover `Example` it is now
```
type Example = {
    name?: string | undefined;
}
```
and `name` is now optional 🧐

Is there something wrong with how I'm bundling the code? Is this a bug in zod?

Context:
zod: 3.21.4
rollup: 3.18.0
typescript: 4.9.3
@rollup/plugin-typescript: 11.0.0

**Comment by JacobWeisenburger** [comment, marked as answer]
are you using `strict` mode? I'm not sure if that would effect it, but just curious.

**Reply by MerlinMason** [reply]
Arrrgh no 🤦🏻 that's fixed it, thanks so much!

**Reply by JacobWeisenburger** [reply]
If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by MerlinMason** [reply]
Yeah good point - I've just raised a question with the engineering leadership team in our company to get an OSS sponsorship budget, hopefully can get something agreed.

**Reply by JacobWeisenburger** [reply]
I'm also open to job opportunities. 😄
